core/shared/src/main/scala/zio/Semaphore.scala

```diff
@@ -65,6 +65,11 @@ sealed trait Semaphore extends Serializable {
    * permits and releasing them when the scope is closed.
    */
   def withPermitsScoped(n: Long)(implicit trace: Trace): ZIO[Scope, Nothing, Unit]
+
+  /**
+   * Returns the number of fibers awaiting for permits.
+   */
+  def awaiting(implicit trace: Trace): UIO[Long]
```

Comment:
Unfortunately this isn't binary compatible. You need to use a stub implementation for it and override it below